pkg/ingester/ingester_v2.go

```diff
@@ -397,7 +456,11 @@ func (i *Ingester) v2Push(ctx context.Context, req *client.WriteRequest) (*clien
 	}
 
 	if firstPartialErr != nil {
-		return &client.WriteResponse{}, httpgrpc.Errorf(http.StatusBadRequest, wrapWithUser(firstPartialErr, userID).Error())
+		code := http.StatusBadRequest
+		if ve, ok := errors.Cause(firstPartialErr).(*validationError); ok {
```

Comment:
I would use `errors.As()` here as well instead of `.(*validationError)` for the same reason suggested by peter.